### PR TITLE
change waf log - add transaction id and mandatory info, fix minor issues

### PR DESCRIPTION
**Pull request by alonzop**

### CODE REVIEW

File: apache2/waf_logging/waf_format.proto [hunk 1]
```diff
@@ -19,19 +19,15 @@ message Properties {
...
-  optional ActionType action = 9;
+  optional string action = 9;
 
   enum SiteType {
     Global = 0;
   }
   optional SiteType site = 10;
   optional Details details = 11;
   optional string hostname = 12;
+  optional string transaction = 13;
```
Comment (meetug) [inline apache2/waf_logging/waf_format.proto]: transactionId?

File: apache2/apache2_util.c [hunk 2]
```diff
@@ -331,19 +332,21 @@ static void send_waf_log(struct waf_lock* lock, apr_file_t* fd, const char* str1
...
+	char waf_unique_id[100] = "";
 
     get_field_value("[file ", "]", str1, waf_filename);
     get_field_value("[id ", "]", str1, waf_id);
     get_field_value("[line ", "]", str1, waf_line);
     get_field_value("[msg ", "]", str1, waf_message);
     get_field_value("[data ", "]", str1, waf_data);
-    get_field_value("[ver ", "]", str1, waf_ruleset_info);
-    get_ip_port(ip_port, waf_ip, waf_port);
+	get_field_value("[ver ", "]", str1, waf_ruleset_info);
```
Comment (meetug) [inline apache2/apache2_util.c:342]: Nit: spacing

File: apache2/waf_logging/waf_log_util.cc [hunk 3]
```diff
@@ -46,25 +49,57 @@ void set_waf_format(waf_format::Waf_Format* waf_format, char* resourceId, char*
...
-            properties->set_action(waf_format::Properties::Detected);
-            break;
-        case 2:
-            properties->set_action(waf_format::Properties::Blocked);
-            break;
-        default:
-            break;
-    }
+	bool is_mandatory = false;
+	if (strcmp(ruleId, RULE_TYPE_OWASP_CRS)) {
```
Comment (meetug) [inline apache2/waf_logging/waf_log_util.cc]: strcmp == 0?

File: apache2/waf_logging/waf_log_util.cc [hunk 4]
```diff
@@ -46,25 +49,57 @@ void set_waf_format(waf_format::Waf_Format* waf_format, char* resourceId, char*
...
+		int tmpid = atoi(ruleId+1);
+		is_mandatory = rule_is_mandatory(tmpid);
+
+		if (ruleSetVersion[0] != '2' && !is_mandatory)
+			properties->set_action(WAF_ACTION_MATCHED);
+		else {
+			switch (action) {
+			case 1:
```
Comment (meetug) [inline apache2/waf_logging/waf_log_util.cc]: define an enum?

**Review by allanbomsft — DISMISSED** [review]

File: apache2/apache2_util.c (hunk 2 above)
Comment (alonzop) [reply apache2/apache2_util.c:342]: it looks fine in my ide....

File: apache2/waf_logging/waf_log_util.cc (hunk 4 above)
Comment (alonzop) [reply apache2/waf_logging/waf_log_util.cc]: sure

**Review by meetug — APPROVED** [review]

**Pull request merged**